File: Meteo_Test_Backend/src/main/java/com/nazzareno/it/model/Meteo.java
package com.nazzareno.it.model;

import java.math.BigDecimal;
import java.time.LocalDate;

import jakarta.persistence.*;


@Entity
@Table(name = "meteo")

public class Meteo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String città;

    @Column(nullable = false)
    private LocalDate data;

    @Column(nullable = false)
    private double tempmax;

    @Column(nullable = false)
    private double tempmin;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getCittà() {
		return città;
	}

	public void setCittà(String città) {
		this.città = città;
	}

	public LocalDate getData() {
		return data;
	}

	public void setData(LocalDate data) {
		this.data = data;
	}

	public double getTempmax() {
		return tempmax;
	}

	public void setTempmax(double tempmax) {
		this.tempmax = tempmax;
	}

	public double getTempmin() {
		return tempmin;
	}

	public void setTempmin(double tempmin) {
		this.tempmin = tempmin;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}
    
}

File: MeteoApp_Frontend/src/app/model/Meteo.ts
export class Meteo {
    public citta: string;
    public data: Date; // LocalDate in Java viene tradotto in Date in TypeScript
    public tempmax: number; // In TypeScript, il tipo `double` è rappresentato da `number`
    public tempmin: number; // In TypeScript, il tipo `double` è rappresentato da `number`
    public userId: number;
  
    constructor(
      citta: string,
      data: Date,
      tempmax: number,
      tempmin: number,
      userId: number
    ) {
      this.citta = citta;
      this.data = data;
      this.tempmax = tempmax;
      this.tempmin = tempmin;
      this.userId = userId;
    }
  }